pkg/controller/vsphere/metrics.go

```diff
@@ -0,0 +1,78 @@
+package vsphere
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	failedInstanceCreateCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "mapi_instance_create_failed",
+			Help: "Number of times provider instance create has failed.",
+		}, []string{"name", "namespace", "reason", "uid", "provider_name"},
+	)
+
+	failedInstanceUpdateCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "mapi_instance_update_failed",
+			Help: "Number of times provider instance update has failed.",
+		}, []string{"name", "namespace", "reason", "uid", "provider_name"},
+	)
+
+	failedInstanceDeleteCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "mapi_instance_delete_failed",
+			Help: "Number of times provider instance delete has failed.",
+		}, []string{"name", "namespace", "reason", "uid", "provider_name"},
+	)
+)
```

Comment:
Were these copied from other providers? /do they match the other providers?

Reply:
Yes, the metrics there are the same as in AWS. Would import those at some point, just need to test it once again, merge it and revendor.